RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,1030 @@
+    int coolingsteps =  1 - int(std::log(T0 * betamax) / std::log(coolingFactor_));
+    std::cout << "DAClustrizerInZT_vect.beta0:   nstep = " << coolingsteps << std::endl;
+  }
+#endif
+
+
+  if (T0 > 1. / betamax) {
+    return betamax / std::pow(coolingFactor_, int(std::log(T0 * betamax) / std::log(coolingFactor_)) - 1);
+  } else {
+    // ensure at least one annealing step
+    return betamax * coolingFactor_;
+  }
+}
+
+
+  
+bool 
+DAClusterizerInZT_vect::split(const double beta,  track_t &tks, vertex_t & y, double threshold ) const{
+  // split only critical vertices (Tc >~ T=1/beta   <==>   beta*Tc>~1)
+  // an update must have been made just before doing this (same beta, no merging)
+  // returns true if at least one cluster was split
+  
+  constexpr double epsilonz=1e-3;      // minimum split size z
+  constexpr double epsilont=1e-2;      // minimum split size t
+  unsigned int nv = y.getSize();
+  
+  // avoid left-right biases by splitting highest Tc first
+  
+  std::vector<std::pair<double, unsigned int> > critical;
+  for(unsigned int k=0; k<nv; k++){    
+    double Tc= 2*y.swE_[k]/y.sw_[k];
+    if (beta*Tc > threshold){
+      critical.push_back( make_pair(Tc, k));
+    }
+  }
+  if (critical.empty()) return false;
+
+
+  std::stable_sort(critical.begin(), critical.end(), std::greater<std::pair<double, unsigned int> >() );
+  
+  
+  bool split=false;
+  const unsigned int nt = tks.getSize();
+
+  for(unsigned int ic=0; ic<critical.size(); ic++){
+    unsigned int k=critical[ic].second;
+
+    // estimate subcluster positions and weight
+    double p1=0, z1=0, t1=0, w1=0;
+    double p2=0, z2=0, t2=0, w2=0;
+    for(unsigned int i=0; i<nt; ++i){
+      if (tks.Z_sum_[i] > 1.e-100) {
+	// winner-takes-all, usually overestimates splitting
+	double tl = tks.z_[i] < y.z_[k] ? 1.: 0.;
+	double tr = 1. - tl;
+
+	// soften it, especially at low T
+	/*
+	double arg = ( tks.z_[i] - y.z_[k] ) * sqrt(beta * tks.dz2_[i]); // + std::fabs(tks.t_[i] - y.t_[k])
+	std::cout << arg << std::endl;
+	if(std::fabs(arg) < 20){
+	  double t = local_exp(-arg);
+	  tl = t/(t+1.);
+	  tr = 1/(t+1.);
+	}
+	*/
+
+	double p = y.pk_[k] * tks.pi_[i] * local_exp(-beta * Eik(tks.z_[i], y.z_[k], tks.dz2_[i], 
+                                                                 tks.t_[i], y.t_[k], tks.dt2_[i])) / tks.Z_sum_[i];
+	double w = p*tks.errsum_[i];
+	p1 += p*tl;  z1 += w*tl*tks.z_[i]; t1 += w*tl*tks.t_[i]; w1 += w*tl;
+	p2 += p*tr;  z2 += w*tr*tks.z_[i]; t2 += w*tr*tks.t_[i]; w2 += w*tr;
+      }
+    }
+
+    if(w1>0){z1 = z1/w1; t1 = t1/w1;} else {z1=y.z_[k]-epsilonz; t1=y.t_[k]-epsilont;}
+    if(w2>0){z2 = z2/w2; t2 = t2/w2;} else {z2=y.z_[k]+epsilonz; t2=y.t_[k]+epsilont;}
+    
+    // reduce split size if there is not enough room
+    if( ( k   > 0 ) && ( z1 < (0.6*y.z_[k] + 0.4*y.z_[k-1])) ){ 
+      z1 = 0.5*y.z_[k] + 0.5*y.z_[k-1]; 
+      t1 = 0.5*y.t_[k] + 0.5*y.t_[k-1]; 
```

Comment:
The Z_vect version has the same constants for k and k-1 both in the condition and in the body of the if.
Is this change justifiable for ZT_vect?
If so, perhaps the uneven choice of constants should be made configurable.

Reply:
Hi Slava, I changed these back to the old way to keep the results consistent with the old clusterizer.

Reply:
I'm not sure what you mean. 
- is there a commit (not pushed yet) in this PR that would revert it back to the way it is in Z_vect ?
- is the current state supposed to be the revert to pre- #18330 state?

If it's the latter, the old version had different values in the condition statement
```C++
    if( ( k   > 0 ) && ( y._z[k-1]>=z1 ) ){ z1=0.5*(y._z[k]+y._z[k-1]); }
```
Given that there are many other functional changes changing behavior wrt old clusterizer, is this "consistency" revert justifiable?